integration-tests/kafka-oauth-keycloak/src/test/java/io/quarkus/it/kafka/KafkaKeycloakTestResource.java

```diff
@@ -18,16 +20,22 @@ public class KafkaKeycloakTestResource implements QuarkusTestResourceLifecycleMa
     private StrimziKafkaContainer kafka;
     private KeycloakContainer keycloak;
 
+    private static final String KEYCLOAK_REALM_JSON = System.getProperty("keycloak.realm.json");
+
     @Override
     public Map<String, String> start() {
 
         Map<String, String> properties = new HashMap<>();
 
         //Start keycloak container
-        keycloak = new KeycloakContainer();
+        keycloak = new KeycloakContainer()
+                .withNetwork(Network.SHARED)
+                .withNetworkAliases("keycloak")
+                .withFixedPort(8080);
         keycloak.start();
         log.info(keycloak.getLogs());
-        keycloak.createHostsFile();
+        createHostsFile(keycloak);
+        keycloak.createRealmFromPath(KEYCLOAK_REALM_JSON);
```

Comment:
@ozangunalp Right now 2 realms are created, the default one `quarkus` (which is not needed for this test) and the one supplied from a classpath here.

IMHO, it would make sense to move `private static final String KEYCLOAK_REALM_JSON = System.getProperty("keycloak.realm.json");` to `KeycloakContainer` (perhaps it should be named `keycloak.realm.path` but if you prefer `keycloak.realm.json` then it is fine as well) and then add `.withEnv(KEYCLOAK_REALM, ...)`, you can copy this code:

https://github.com/quarkusio/quarkus/blob/main/extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevServicesProcessor.java#L442

(minus keycloakX specific branches for now).

And then update `KeycloakTestResourceLifecycleManager` to skip adding a default realm if `keycloak.realm.json` (or `keycloak.realm.path`) is set.

IMHO it would be more optimal

Reply:
@sberyozkin I don't think that the default realm is created for this test. It doesn't use the `KeycloakTestResourceLifecycleManager`.

Reply:
@ozangunalp I missed, sorry, but since you have moved the container to that module, would it not make for individual tests not to worry about loading the realm from the disk since the system property is checked here anyway, with `KeycloakContainer` doing it instead ? I think this PR adding an option to use a custom realm is great, but it does not help much other `KeycloakTestResourceLifecycleManager` users